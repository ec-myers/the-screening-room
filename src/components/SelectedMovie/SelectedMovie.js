import React from 'react';
import './SelectedMovie.scss';
import { Link } from 'react-router-dom';

const SelectedMovie = ({ movieDetails, wallpapers }) => {
  const { title, overview, poster, rating, id } = movieDetails;
  const movieWallpaper = wallpapers.find(wallpaper => wallpaper.id === id).wallpaper
  console.log(movieWallpaper)
  return (
    <>
      <Link to='/'>
        <div className='modal-backdrop-white'>
          <div className='modal-backdrop'></div>
        </div>
      </Link>
      <main className='SelectedMovie movie-modal'>
        <div className='SelectedMovie__div--container'>
          <header className='SelectedMovie__header'>
            <img
              className='SelectedMovie__img--favorite'
              src={movieWallpaper}
              alt='Favorite icon'
            />
            <img
              className='SelectedMovie__img--close'
              src={poster}
              alt='Close window icon'
            />
          </header>
          <div className='bottom-container'>
            <div className='movie-text'>
              <h2>{title}</h2>
              <h3 className='SelectedMovie__h3'>{overview}</h3>
            </div>
            <div className='movie-details'>
              <h4>
                Rating: <span>{rating}</span>
              </h4>
            </div>
          </div>
        </div>
      </main>
    </>
  );
};

export default SelectedMovie;
