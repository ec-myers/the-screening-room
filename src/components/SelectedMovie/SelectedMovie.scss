.SelectedMovie__img--favorite {
  width: 100%;
  position: relative;
  -webkit-filter: brightness(20%);
  filter:brightness(68%);
  border-radius: 4px 4px 0 0;
  box-shadow: 0 3px 3px rgba(0, 0, 0, 0.26);
  object-fit: cover;
  height: 400px;
}

.SelectedMovie__img--close {
  width: 200px;
  position: absolute;
  bottom: 100px;
  left: 32px;
  box-shadow: 0 3px 10px rgba(0, 0, 0, 0.747);
}

.movie-modal {
  position: fixed;
  z-index: 100;
  top: 50%;
  left: 50%;
  transform: translate3d(-50%, -50%, 0);
  display: flex;
  flex-direction: column;
  background-color: white;
  border-radius: 4px;
  border: none;
  box-shadow: 0 2px 10px rgba(0, 0, 0, 0.301);
  min-width: 20rem;
  max-width: 60rem;
  width: 800px;
  height: 650px;
  background: rgb(65, 65, 65);

  h2 {
    color: rgb(230, 230, 230);
    font-size: 30px;
    z-index: 1;
  }

  h3 {
    margin-top: 30px;
    font-size: 17px;
    color: rgb(212, 212, 212);
    font-weight: 400;
  }

  .movie-details {
    display: flex;
    flex-direction: column;
    margin-left: 32px;
    position: relative;
    top: 160px;
    color: rgb(175, 175, 175);
    font-weight: 300;
  }

  h4 {
    color: rgb(175, 175, 175);
    font-weight: 400;
    padding-bottom: 5px;
    width: 200px;

    span {
      font-weight: 300;
    }
  }

}

.movie-text {
  width: 65%;
  position: relative;
  margin-right: 50px;
  left: 30px;
  bottom: 50px;
}

.bottom-container {
  display: flex;
  justify-content: space-between;
  flex-direction: row-reverse;
}