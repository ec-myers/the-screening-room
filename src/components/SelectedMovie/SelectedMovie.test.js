import React from 'react';
import { shallow } from 'enzyme';
import SelectedMovie from './SelectedMovie';

describe('SelectedMovie', () => {
  let wrapper;
  let movieDetails = {
    wallpaper:
      'https://image.tmdb.org/t/p/original/skvI4rYFrKXS73BJxWGH54Omlvv.jpg',
    poster:
      'https://image.tmdb.org/t/p/original/tBuabjEqxzoUBHfbyNbd8ulgy5j.jpg',
    title: 'Maleficent: Mistress of Evil',
    overview:
      'Maleficent and her goddaughter Aurora begin to question the complex family ties that bind them as they are pulled in different directions by impending nuptials, unexpected allies, and dark new forces at play.',
    genre: ['Adventure', 'Fantasy', 'Family'],
    rating: 7.2
  };

  beforeEach(() => {
    wrapper = shallow(<SelectedMovie movieDetails={movieDetails} />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
