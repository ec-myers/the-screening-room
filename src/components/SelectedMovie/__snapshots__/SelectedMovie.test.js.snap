// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectedMovie should match the snapshot 1`] = `
<Fragment>
  <Link
    to="/"
  >
    <div
      className="modal-backdrop-white"
    >
      <div
        className="modal-backdrop"
      />
    </div>
  </Link>
  <main
    className="SelectedMovie movie-modal"
  >
    <div
      className="SelectedMovie__div--container"
    >
      <header
        className="SelectedMovie__header"
      >
        <img
          alt="Favorite icon"
          className="SelectedMovie__img--favorite"
          src="https://image.tmdb.org/t/p/original/skvI4rYFrKXS73BJxWGH54Omlvv.jpg"
        />
        <img
          alt="Close window icon"
          className="SelectedMovie__img--close"
          src="https://image.tmdb.org/t/p/original/tBuabjEqxzoUBHfbyNbd8ulgy5j.jpg"
        />
      </header>
      <div
        className="bottom-container"
      >
        <div
          className="movie-text"
        >
          <h2>
            Maleficent: Mistress of Evil
          </h2>
          <h3
            className="SelectedMovie__h3"
          >
            Maleficent and her goddaughter Aurora begin to question the complex family ties that bind them as they are pulled in different directions by impending nuptials, unexpected allies, and dark new forces at play.
          </h3>
        </div>
        <div
          className="movie-details"
        >
          <h4>
            Rating: 
            <span>
              7.2
            </span>
          </h4>
        </div>
      </div>
    </div>
  </main>
</Fragment>
`;
